### PR TITLE
Integrate JWT Authentication and Route Verification

### PR DESCRIPTION
**Pull request by samar12-rad**
Description:

What: Integrated JSON Web Token (JWT) authentication for user login and protected routes.
Why: To enhance security by ensuring that only authenticated users can access certain routes.

Changes Made:
Implemented JWT for user login and token generation.
Created middleware for JWT verification.
Protected sensitive routes with JWT verification.
Updated existing login and signup routes to include JWT generation.

Testing:
Verified that users can log in and receive a JWT.
Ensured protected routes return the correct responses when accessed with valid and invalid tokens.

Related Issues:

[#146 ]

## Release Notes

- **New Features**
  - Introduced JWT authentication for admin and customer login, enhancing security.
  - Added password reset functionality for customers.
  - New `ResetPassword` component for user password management.

- **Improvements**
  - Updated authentication middleware for admin and customer routes.
  - Enhanced login responses to include user roles and tokens.
  - Updated the way environment variables are accessed in the `Login` component.
  - Added a new route for the `ResetPassword` component in the routing configuration.

- **Bug Fixes**
  - Improved error handling and validation during login and password reset processes.

### CODE REVIEW

**Comment by samar12-rad** [comment]
@RamakrushnaBiswal , why is it showing i want to merge 8 commits, i dont want that, it is giving unnecessary coderabbit error?

**Comment by RamakrushnaBiswal** [comment]
resolve conflicts @samar12-rad

**Comment by samar12-rad** [comment]
Ok

File: backend/middlewares/authAdmin.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      req.user = admin;
+      logger.info(`Admin authenticated: ${JSON.stringify(admin.email)}`);
+      next();
+    });
+  } else {
+    res.sendStatus(401); // Unauthorized
+  }
+};
+
+module.exports = authenticateAdmin;
```
[the remark this answers is not in the document]
Comment (samar12-rad) [reply backend/middlewares/authAdmin.js:24]: @RamakrushnaBiswal , can you set JWT_SECRET in production environment pls?

File: backend/middlewares/authAdmin.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      }
+
+      req.user = decoded;
+      logger.info(`Admin authenticated: ${JSON.stringify(decoded.email)}`);
+      next();
+    });
+  } else {
+    res.sendStatus(401); // Unauthorized
+  }
+};
```
[the remark this answers is not in the document]
Comment (samar12-rad) [reply backend/middlewares/authAdmin.js:23]: Now jwt doesnt sign email in body. So not needed

**Comment by RamakrushnaBiswal** [comment]
@samar12-rad resolve the conflicts :)

**Comment by samar12-rad** [comment]
Resolving merge conflicts

**Comment by samar12-rad** [comment]
resolved branch conflicts @RamakrushnaBiswal

**Pull request merged**